keras/layers/wrappers.py

```diff
@@ -227,6 +246,59 @@ def step(x, _):
             y._uses_learning_phase = True
         return y
 
+    def compute_mask(self, inputs, mask=None):
+        # cases need to call the layer.compute_mask when input_mask is None:
+        # Masking layer and Embedding layer with mask_zero
+
+        def get_shape_tuple(tensor, start_idx, init_tuple, int_shape=None):
+            # replace all None in int_shape by K.shape
+            if int_shape is None:
+                int_shape = K.int_shape(tensor)[start_idx:]
+            if not any(not s for s in int_shape):
+                return init_tuple + int_shape
+            tensor_shape = K.shape(tensor)
+            int_shape = list(int_shape)
+            for i, s in enumerate(int_shape):
+                if not s:
+                    int_shape[i] = tensor_shape[start_idx + i]
+            return init_tuple + tuple(int_shape)
+
+        input_shape = K.int_shape(inputs)
+        if input_shape[0]:
+            # batch size matters, we currently do not handle mask explicitly
+            return mask
+        inner_mask = mask
+        if inner_mask is not None:
+            inner_mask_shape = get_shape_tuple(mask, 2, (-1,))
+            inner_mask = K.reshape(inner_mask, inner_mask_shape)
+        input_uid = object_list_uid(inputs)
+        inner_inputs = self._input_map[input_uid]
+        output_mask = self.layer.compute_mask(inner_inputs, inner_mask)
+        if output_mask is None:
+            if mask is None:
+                return None
+            # input_mask is not None, and output_mask is None: we should return a not-None mask
+            output_mask = mask
+            for _ in range(2, len(K.int_shape(mask))):
+                output_mask = K.any(output_mask, axis=-1)
+        else:
+            # output_mask is not None. We need to reshape it
+            input_length = input_shape[1]
+            if not input_length:
+                input_length = K.shape(inputs)[1]
+            output_mask_int_shape = K.int_shape(output_mask)
+            if output_mask_int_shape is None:
+                # if the output_mask does not have `_keras_shape`,
```

Comment:
No reference to `_keras_shape`, that's an implementation detail. Call it "static shape"